homework/index.js

```diff
@@ -23,24 +10,151 @@
+    class: 'page-title',
+    text: 'HYF Repositories',
+  });
+  const selectEl = createAndAppend('select', header, {
+    class: 'select-list',
+    id: 'repositories_list',
+  });
+  const infoContainer = createAndAppend('div', root, {
+    id: 'info_container',
+    class: 'info-container',
+  });
+
+  function renderError(error) {
+    createAndAppend('h1', root, { text: error.message, class: 'alert-error' });
+  }
+
+  function fetchJSON(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+      xhr.onload = () => {
+        if (xhr.status >= 200 && xhr.status <= 299) {
+          resolve(xhr.response);
+        } else {
+          reject(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        }
+      };
+      xhr.onerror = () => reject(new Error('Network request failed'));
+      xhr.send();
+    });
+  }
+
+  function showContributions(urlText) {
+    const infoRight = createAndAppend('div', infoContainer, {
+      class: 'info-right',
+      text: 'Contributions:',
+    });
+
+    function renderContributions(contributors) {
+      contributors.forEach(contributor => {
+        const contributorLink = createAndAppend('a', infoRight, {
+          class: 'contributor-link',
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        const imgDiv = createAndAppend('div', contributorLink, { class: 'contributor-img-div' });
+        createAndAppend('img', imgDiv, {
+          class: 'contributor-img',
+          src: contributor.avatar_url,
+          alt: 'Contributor photo',
+        });
+        imgDiv.firstChild.style.maxHeight = '50px';
+        const dataDiv = createAndAppend('div', contributorLink, { class: 'contributor-data-div' });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
+      if (infoRight.firstElementChild === null) {
+        createAndAppend('p', infoRight, {
+          class: 'no-content',
+          text: 'There are no contributions yet.',
+        });
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+    }
+
+    fetchJSON(urlText)
+      .then(goodResult => renderContributions(goodResult))
+      .catch(somethingWrong => renderError(somethingWrong));
+  }
+
+  function showRepoInfo(wantedRepo) {
+    const infoLeft = createAndAppend('div', infoContainer, {
+      class: 'info-left',
+      text: 'Repository information:',
     });
+    const table = createAndAppend('table', infoLeft);
+    const tbody = createAndAppend('tbody', table);
+    addRow(tbody, `Name`, wantedRepo.name);
```

Comment:
With `addRow()` returning the `tr` it created, you can now assign its return value to a variable, e.g.:

```js
const firstRow = addRow(tbody, `Name`, wantedRepo.name);
```

Reply:
I made a mistake here. The third argument should just be an empty string ‘’ because you are going to add a child anchor tag.

Reply:
Thanks